scripts/mbedtls_framework/build_tree.py

```diff
@@ -122,3 +129,13 @@ def guess_tf_psa_crypto_root(root: Optional[str] = None) -> str:
         return root
     else:
         raise Exception('TF-PSA-Crypto source tree not found')
+
+def is_mbedtls_3_6_branch(root: Optional[str] = None) -> bool:
+    """Whether root is the root directory of the Mbed TLS 3.6 branch """
+    if root is None:
+        root = os.getcwd()
+    if not looks_like_mbedtls_root(root):
+        return False
+    with open(os.path.join(root, 'include', 'mbedtls', 'build_info.h'), 'r') as f:
```

Comment:
There is no error handling or documentation of what will happen if the build_info cannot be found and open will throw an exception.

Reply:
Yes this was intended. The `build_info.h` path is hardcoded in the function thus I prefer to stop immediately in this function if the path is not correct anymore.